src/main/java/uk/gov/companieshouse/api/accounts/service/impl/DebtorsService.java

```diff
@@ -0,0 +1,128 @@
+import uk.gov.companieshouse.api.accounts.links.TransactionLinkType;
+import uk.gov.companieshouse.api.accounts.model.entity.notes.debtors.DebtorsEntity;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.Debtors.Debtors;
+import uk.gov.companieshouse.api.accounts.repository.DebtorsRepository;
+import uk.gov.companieshouse.api.accounts.service.ResourceService;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseStatus;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.api.accounts.transformer.DebtorsTransformer;
+import uk.gov.companieshouse.api.accounts.utility.impl.KeyIdGenerator;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+public class DebtorsService implements ResourceService<Debtors> {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(CompanyAccountsApplication.APPLICATION_NAME_SPACE);
+
+    private DebtorsRepository repository;
+    private DebtorsTransformer transformer;
+    private SmallFullService smallFullService;
+    private KeyIdGenerator keyIdGenerator;
+
+    @Autowired
+    public DebtorsService(DebtorsRepository repository,
+                          DebtorsTransformer transformer,
+                          SmallFullService smallFullService,
+                          KeyIdGenerator keyIdGenerator) {
+
+        this.repository = repository;
+        this.transformer = transformer;
+        this.smallFullService = smallFullService;
+        this.keyIdGenerator = keyIdGenerator;
+    }
+
+    @Override
+    public ResponseObject<Debtors> create(Debtors rest, Transaction transaction,
+                                          String companyAccountsId, HttpServletRequest request)
+            throws DataException {
+
+        setMetadataOnRestObject(rest, transaction, companyAccountsId);
+
+        DebtorsEntity entity = transformer.transform(rest);
+        entity.setId(generateID(companyAccountsId));
+
+        try {
+
+            repository.insert(entity);
+        } catch (DuplicateKeyException e) {
+
+            LOGGER.errorRequest(request, e, getDebugMap(transaction, companyAccountsId, entity.getId()));
+
+            return new ResponseObject<>(ResponseStatus.DUPLICATE_KEY_ERROR);
+        } catch (MongoException e) {
+
+            DataException dataException = new DataException("Failed to insert " + ResourceName.DEBTORS.getName(), e);
+            LOGGER.errorRequest(request, dataException, getDebugMap(transaction, companyAccountsId, entity.getId()));
+
+            throw dataException;
+        }
+
+        smallFullService
+                .addLink(companyAccountsId, SmallFullLinkType.DEBTORS_NOTE,
+                        getSelfLinkFromDebtors(entity), request);
```

Comment:
we could rename this to `getSelfLinkFromDebtorsEntity` to distinguish it from the another `Debtors rest`